pkg/container/docker_pull.go

```diff
@@ -65,6 +73,30 @@ func NewDockerPullExecutor(input NewDockerPullExecutorInput) common.Executor {
 	}
 }
 
+func getImagePullOptions(ctx context.Context, input NewDockerPullExecutorInput) (types.ImagePullOptions, error) {
+	imagePullOptions := types.ImagePullOptions{
+		Platform: input.Platform,
+	}
+	if input.Username != "" && input.Password != "" {
+		logger := common.Logger(ctx)
+		logger.Debugf("  using authentication")
```

Comment:
Perhaps the debug message could be a little bit more descriptive since in log it's quite vague

Reply:
Our first attempt was to add at least the username, but since that is (as well as the password) a secret, it is replaced by `***`.
Therefore we removed that, because we find it useless.
Any suggestions what to log?

Reply:
e.g.
```suggestion
		logger.Debugf("using authentication for docker pull")
```

Reply:
At least it is clear that the codepath with authentiction was taken (if it fails).